### PR TITLE
Fix 4:00 - 5:00 pm interpreted as 4am to 5pm

### PR DESCRIPTION
**Pull request by STayinloves**
GitHub issue https://github.com/Microsoft/Recognizers-Text/issues/386

### CODE REVIEW

File: Specs/DateTime/English/TimePeriodParser.json [hunk 1]
```diff
@@ -1027,4 +1027,160 @@
...
+{
+  "Input": "please help me set up a meeting from 1:30 to 3pm",
+  "Context": {
+    "ReferenceDateTime": "2017-12-01T13:37:00"
+  },
+  "NotSupported": "javascript",
+  "NotSupportedByDesign": "python",
+  "Results": [
+    {
+      "Text": "from 1:30 to 3pm",
```
Comment (yiylin) [inline Specs/DateTime/English/TimePeriodParser.json]: do we support "from 1:30am to 3"? if we support, please add these kind of test case

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseTimePeriodParser.cs [hunk 2]
```diff
@@ -137,15 +137,20 @@ private DateTimeResolutionResult ParseSimpleCases(string text, DateObject refere
...
+                        }
+
+                        if (beginHour >= 12 && beginHour - 12 < endHour)
                         {
                             beginHour -= 12;
                         }
 
-                        if (endHour >= 12)
+                        // Resolve case like "11 to 3am"
+                        if (beginHour < 12 && beginHour > endHour)
```
Comment (yiylin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseTimePeriodParser.cs]: It is better that the PR title include the Issue number. Then I can check the details of the bug

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseTimePeriodParser.cs]: In the description, not in the title. :)

File: Specs/DateTime/English/TimePeriodParser.json [hunk 3]
```diff
@@ -899,6 +899,32 @@
   ]
 }
 ,
+{
+  "Input": "please help me set up a meeting from 1:30am to 3",
```
Comment (tellarin) [inline Specs/DateTime/English/TimePeriodParser.json]: Can we also have the other cases?
If possible, please write different senteces too.

File: Specs/DateTime/English/TimePeriodParser.json (hunk 1 above)
Comment (yiylin) [reply Specs/DateTime/English/TimePeriodParser.json]: I prefer to add more cases. "from 11am to 2", "from 11pm to 1", "from 0:01am to 1pm", "from 0:01am to 1". Not sure if some of them has been added

Comment (STayinloves) [reply Specs/DateTime/English/TimePeriodParser.json]: Previously I regarded that the error only occurred when "am/pm" is a suffix, now I have fixed all cases and add tests.

**Comment by tellarin** [comment]
Thanks, @STayinloves.

**Pull request merged**